GVRf/Framework/framework/src/main/jni/engine/renderer/batch.cpp

```diff
@@ -164,68 +160,52 @@ bool Batch::isRenderDataDisabled(){
     }
     return update_vbo;
 }
-void Batch::setupMesh(){
-    bool update_vbo = isRenderDataDisabled();
-
-    if(render_data_set_.size() ==0)
-        return;
-
-    if(update_vbo)
-        regenerateMeshData();
+void Batch::resetBatch(){
+    clearData();
+    delete renderdata_;
```

Comment:
Wouldn't it be safer if renderdata_ was also set to nullptr? In the future the dtor may actually execute and the flow may be such that this renderdata_ may be deleted second time.